pkg/network/config/config_test.go

```diff
@@ -444,9 +444,51 @@ func TestHTTPReplaceRules(t *testing.T) {
 		},
 	}
 
+	envContent := `
+        [
+          {
+            "pattern": "/users/(.*)",
+            "repl": "/users/?"
+          },
+          {
+            "pattern": "foo",
+            "repl": "bar"
+          },
+          {
+            "pattern": "payment_id"
+          }
+        ]
+        `
+
+	t.Run("via deprecated YAML", func(t *testing.T) {
+		newConfig(t)
+		_, err := sysconfig.New("./testdata/TestDDSystemProbeConfig-HTTPReplaceRulesDeprecated.yaml")
+		require.NoError(t, err)
+		cfg := New()
+
+		require.Len(t, cfg.HTTPReplaceRules, 3)
+		for i, r := range expected {
+			assert.Equal(t, r, cfg.HTTPReplaceRules[i])
+		}
+	})
+
+	t.Run("via deprecated ENV variable", func(t *testing.T) {
+		newConfig(t)
+		t.Setenv("DD_SYSTEM_PROBE_NETWORK_HTTP_REPLACE_RULES", envContent)
+
+		_, err := sysconfig.New("")
+		require.NoError(t, err)
```

Comment:
If you are just testing with environment variables, you shouldn't need to call `sysconfig.New` now.